### DISCUSSION TITLE
Delete data by removing files from disk

### DISCUSSION THREAD

**Discussion by fatecyx** [Difegue/LANraragi#902] [Q&A] [answered]
Some APIs return data that has already been deleted, 
Should a button be added in the settings to scan the database and delete data that files not exists?

**Comment by fatecyx** [comment]
when a file was deleted, I saw in the log detected it. 
Is there no change in this API (/api/archives/untagged) data source?

**Comment by Difegue** [comment, marked as answer]
The Clean Database button does that.  
![image](https://github.com/Difegue/LANraragi/assets/8237712/a5d3e4f6-d859-461e-a9c4-e89fb65fb82c)

**Comment by fatecyx** [comment]
I tried to execute this' clean database ', but it didn't work.

This API (/api/archives/untagged) still returns a large amount of data, and I have tagged all the files

**Comment by Difegue** [comment]
In some cases clean will only unlink filesystem entries first before actually removing them -- Have you checked the API response? It should show counts for both deleted and unlinked entries.

**Comment by fatecyx** [comment]
oh...it seems has been solved...

yesterday, i excuted "clean database" and it clear some entries (not zero). then in the batch page, the api returns 3000+ results.
but now, I just executed it. it clear 0 entries, and the api returns 0 result

then I delete another file. 
and execute "clean database" , return: {"deleted":1,"operation":"clean_database","success":1,"unlinked":0}
and the api return 1 result
but after some minutes, the api returns 0

i think it actually works

**Reply by cazendium** [reply]
In my experience, the first time detects it, then the second time deletes it

**Comment by Difegue** [comment]
There's some delay as the statistic indexes are rebuilt after you hit clean DB - During that time you might indeed still see some outdated data.  
I didn't think it'd be that noticeable though, probably yet another thing that gets worse as you have more files in storage 🤔  

I'll move this to a discussion as there's no real bug here 👋
